### DISCUSSION TITLE
FraggleSim.GetFraggleSimilarity fail

### DISCUSSION THREAD

**Discussion by tzyii** [rdkit/rdkit#6474] [Q&A]
mol1, mol2 = Chem.MolFromSmiles('OCCc1ccn2cnccc12'), Chem.MolFromSmiles('C1CC1Oc1cc2ccncn2c1')
Draw.MolsToGridImage([mol1, mol2], molsPerRow=3, subImgSize=(150, 150))
FraggleSim.GetFraggleSimilarity(mol1, mol2)

output:
[14:35:35] Can't kekulize mol.  Unkekulized atoms: 5 6 7 8 9

---------------------------------------------------------------------------
ArgumentError                             Traceback (most recent call last)
Cell In[27], line 1
----> 1 FraggleSim.GetFraggleSimilarity(mol1, mol2)

File ~/.local/lib/python3.10/site-packages/rdkit/Chem/Fraggle/FraggleSim.py:381, in GetFraggleSimilarity(queryMol, refMol, tverskyThresh)
    379 bestMatch = None
    380 for frag in frags:
--> 381   _, fragsim = compute_fraggle_similarity_for_subs(refMol, queryMol, qSmi, frag, tverskyThresh)
    382   if fragsim > result:
    383     result = fragsim

File ~/.local/lib/python3.10/site-packages/rdkit/Chem/Fraggle/FraggleSim.py:329, in compute_fraggle_similarity_for_subs(inMol, qMol, qSmi, qSubs, tverskyThresh)
    327   qmMolFp = modified_query_fps[qm_key]
    328 else:
--> 329   qmMol = atomContrib(qSubs, qMol, tverskyThresh)
    330   qmMolFp = Chem.RDKFingerprint(qmMol, **rdkitFpParams)
    331   modified_query_fps[qm_key] = qmMolFp

File ~/.local/lib/python3.10/site-packages/rdkit/Chem/Fraggle/FraggleSim.py:277, in atomContrib(subs, mol, tverskyThresh)
    275 # generate fp of query_substructs
    276 qsMol = Chem.MolFromSmiles(subs)
--> 277 subsFp = Chem.RDKFingerprint(qsMol, **rdkitFpParams)
    279 # loop through atoms of smiles get atoms that have a high similarity with substructure
    280 marked = set()

ArgumentError: Python argument types in
    rdkit.Chem.rdmolops.RDKFingerprint(NoneType)
did not match C++ signature:
    RDKFingerprint(RDKit::ROMol mol, unsigned int minPath=1, unsigned int maxPath=7, unsigned int fpSize=2048, unsigned int nBitsPerHash=2, bool useHs=True, double tgtDensity=0.0, unsigned int minSize=128, bool branchedPaths=True, bool useBondOrder=True, boost::python::api::object atomInvariants=0, boost::python::api::object fromAtoms=0, boost::python::api::object atomBits=None, boost::python::api::object bitInfo=None)

**Comment by LingjieBao1998** [comment]
```
  # generate fp of query_substructs
  try:
    qsMol = Chem.MolFromSmiles(subs)
    subsFp = Chem.RDKFingerprint(qsMol, **rdkitFpParams)
  except Exception as e:
    qsMol = Chem.MolFromSmarts(subs)
    subsFp = Chem.RDKFingerprint(qsMol, **rdkitFpParams)
```
This error caused by fragment leading to invalid mol in rdkit when original mol pocess **five-membered heterocyclic aromatic ring**. You can read from Smart to avoid this error.
